tests/core/posix.c

```diff
@@ -27,6 +31,17 @@ void test_core_posix__initialize(void)
 #endif
 }
 
+void test_core_posix__cleanup(void)
+{
+	int i;
+
+	for (i = 0; i < LC_ALL; i++) {
+		if (old_locales[i])
+		    setlocale(LC_COLLATE, old_locales[i]);
```

Comment:
Wait, this is wrong, isn't it? This should rather be `setlocal(i, old_locales[i])`. Otherwise, we repeatedly set `LC_COLLATE`